test/tests/concrete_moisture_heat_transfer/tests

```diff
@@ -9,47 +9,596 @@
       input = 'concrete_thermal_moisture_properties.i'
-      cli_args = 'Materials/concrete/thermal_conductivity_model=EUROCODE-2004 Materials/concrete/thermal_capacity_model=EUROCODE-2004 Outputs/file_base=concrete_thermal_moisture_properties_eurocode_2004_out'
-      csvdiff = 'concrete_thermal_moisture_properties_eurocode_2004_out.csv'
+      csvdiff = 'eurocode_thermal_model_out.csv'
+      cli_args = 'Materials/concrete/thermal_model=EUROCODE-2004 '
+                'Materials/concrete/ref_density=2231 '
+                'Materials/concrete/ref_specific_heat=1100 '
+                'Materials/concrete/moisture_model=Bazant '
+                'Materials/concrete/D1=3.0e-10 '
+                'Materials/concrete/n=16 '
+                'Materials/concrete/critical_relative_humidity=0.75 '
+                'Materials/concrete/coupled_moisture_diffusivity_factor=1e-5 '
+                'Outputs/file_base=eurocode_thermal_model_out '
+      detail = 'Eurocode-2004 thermal model.'
+    []
+    [eurocode_thermal_ref_density]
+      type = 'RunException'
+      input = 'concrete_thermal_moisture_properties.i'
+      expect_err = "For EUROCODE-2004 thermal model, ref_density must be defined."
+      cli_args = 'Materials/concrete/thermal_model=EUROCODE-2004 '
+                'Materials/concrete/ref_specific_heat=1100 '
+                'Materials/concrete/moisture_model=Bazant '
+                'Materials/concrete/D1=3.0e-10 '
+                'Materials/concrete/n=16 '
+                'Materials/concrete/critical_relative_humidity=0.75 '
+                'Materials/concrete/coupled_moisture_diffusivity_factor=1e-5 '
+                'Outputs/file_base=eurocode_thermal_model_out '
+      detail = 'Eurocode-2004 thermal model shall generate error if ref_density is not set by user.'
+    []
+    [eurocode_thermal_ref_specific_heat]
+      type = 'RunException'
+      input = 'concrete_thermal_moisture_properties.i'
+      expect_err = "For EUROCODE-2004 thermal model, ref_specific_heat must be defined."
+      cli_args = 'Materials/concrete/thermal_model=EUROCODE-2004 '
+                'Materials/concrete/ref_density=2231 '
+                'Materials/concrete/moisture_model=Bazant '
+                'Materials/concrete/D1=3.0e-10 '
+                'Materials/concrete/n=16 '
+                'Materials/concrete/critical_relative_humidity=0.75 '
+                'Materials/concrete/coupled_moisture_diffusivity_factor=1e-5 '
+                'Outputs/file_base=eurocode_thermal_model_out '
+      detail = 'Eurocode-2004 thermal model shall generate error if ref_specific_heat is not set by user.'
     []
   []
-  [moisture_properties]
-    issues = '#9'
-    requirement = 'Blackbear shall compute material properties for moisture transport in concrete '
-    [bazant]
-      detail = 'using the Bazant model, '
+
+  [moisture_model]
+    requirement = 'Blackbear shall provide the moisture capacity and moisture diffusivity in domain as demonstrated by'
```

Comment:
Blackbear shall compute the moisture capacity (where applicable) and moisture diffusivity for concrete for the following models and options: